### PR TITLE
Update fn_loadMarkers.sqf

### PR DESCRIPTION
**Pull request by nomisum**

### CODE REVIEW

File: addons/saveMarkers/functions/fn_loadMarkers.sqf [hunk 1]
```diff
@@ -6,45 +6,50 @@ params ["_saveName","_mapName",["_markersData",[]]];
...
-    _marker setMarkerDir _dir;
-    _marker setMarkerPos _pos;
-    _marker setMarkerShape _shape;
-    _marker setMarkerSize _size;
-    _marker setMarkerText _text;
-    _marker setMarkerType _type;
-    if (count _polyline > 0) then {
-        _marker setMarkerPolyline _polyline;
-    };
+        private _marker = createMarkerLocal [_markerID,_pos,_channel,player]; // create locally and broadcast with last command
```
Comment (Fusselwurm) [inline addons/saveMarkers/functions/fn_loadMarkers.sqf:39]: (holy crap, that is even explicit advice in the biki https://community.bistudio.com/wiki/setMarkerDir )

**Comment by DerZade** [comment]
How about letting everyone create their own markers? As long as they have the same name it is even synchronized.

I just tested the following with global exec. Moving and deleting markers works just fine. 
```sqf
"|_USER_DEFINED #962500445/0/0|[100,100,0]|mil_pickup|ICON|[1,1]|0|Solid|Default|1|You are here" call BIS_fnc_stringToMarkerLocal
```

File: addons/saveMarkers/functions/fn_loadMarkers.sqf (hunk 1 above)
Comment (nomisum) [reply addons/saveMarkers/functions/fn_loadMarkers.sqf:39]: since oct 2020, so way after we all learned those commands :D

**Comment by nomisum** [comment]
> BIS_fnc_stringToMarkerLocal

Certainly more elegant if one larger broadcast is more handleable than multiple small. We cant really test either of them without inviting 80 people i guess :P

**Comment by McDiod** [comment]
Does stringToMarker/markerToString support polylines?

Creating markers locally means you'd have to add them to the remoteExec JIP queue. Not sure which is better or worse performance wise.

**Comment by DerZade** [comment]
> Does stringToMarker/markerToString support polylines?

I would hope so, but I'm not certain.


> Creating markers locally means you'd have to add them to the remoteExec JIP queue. Not sure which is better or worse performance wise.

That is indeed a valid point. 🤔
I don't even know how deleted markers would be handled with JIP.

File: addons/saveMarkers/functions/fn_loadMarkers.sqf [hunk 2]
```diff
@@ -6,45 +6,50 @@ params ["_saveName","_mapName",["_markersData",[]]];
...
 
-    // increment marker index until unused _markerID is found
-    private _markerID = format ["_USER_DEFINED #%1/%2/%3",_playerID,_markerIDIndex,_channel];
-    while {markerShape _markerID != ""} do {
-        _markerIDIndex = _markerIDIndex + 1;
-        _markerID = format ["_USER_DEFINED #%1/%2/%3",_playerID,_markerIDIndex,_channel];
-    };
+        // increment marker index until unused _markerID is found
+        private _markerID = format ["_USER_DEFINED #%1/%2/%3",
+        ,_markerIDIndex,_channel];
```
Comment (Fusselwurm) [inline addons/saveMarkers/functions/fn_loadMarkers.sqf:33]: ⚡ b0rked

**Review by Fusselwurm — CHANGES_REQUESTED** [review]

**Comment by Fusselwurm** [comment]
* fixed code
* for better profiling, change the loop thing from `spawn` to `cba_fnc_waitAndExecute`

**Review by Fusselwurm — APPROVED** [review]

**Comment by Fusselwurm** [comment]
> > BIS_fnc_stringToMarkerLocal
> 
> Certainly more elegant if one larger broadcast is more handleable than multiple small. We cant really test either of them without inviting 80 people i guess :P

wouldnt even need to be a broadcast, if handled like the replay - dont give the marker package to everyone all at once, but spread them out with half-second delays or something. efficient & safe much.

**Comment by Fusselwurm** [comment]
that being said, "perfect" is the enemy of "good". 

the optimization that we have here (one marker broadcast per second) is probably good enough. :shipit:

File: addons/saveMarkers/functions/fn_loadMarker.sqf [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+_marker setMarkerColorLocal _color;
+_marker setMarkerDirLocal _dir;
+_marker setMarkerPosLocal _pos;
+_marker setMarkerShapeLocal _shape;
+_marker setMarkerSizeLocal _size;
+_marker setMarkerTextLocal _text;
+_marker setMarkerType _type;
+if (count _polyline > 0) then {
+	_marker setMarkerPolyline _polyline;
+};
```
Comment (McDiod) [inline addons/saveMarkers/functions/fn_loadMarker.sqf:42]:
```suggestion
if (count _polyline > 0) then {
	_marker setMarkerPolylineLocal _polyline;
};
_marker setMarkerType _type;
```

**Pull request merged**